src/System.Data.SqlClient/src/System/Data/SqlClient/SqlDependencyListener.cs

```diff
@@ -612,7 +612,9 @@ private void Restart(object unused)
                         _timeoutParam.Value = _defaultWaitforTimeout; // If success, reset to default for re-queue.
                         AsynchronouslyQueryServiceBrokerQueue();
                         _errorState = false;
+                        Timer retryTimer = _retryTimer;
                         _retryTimer = null;
+                        retryTimer?.Dispose();
```

Comment:
It's not a big deal, but as long as we're doing the null check, we may as well avoid the unnecessary null write:
```C#
Timer retryTimer = _retryTimer;
if (retryTimer != null)
{
    _returnTimer = null;
    retryTimer.Dispose();
}
```

Reply:
Changed